static/app/views/settings/organizationIntegrations/configureIntegration.tsx

```diff
@@ -166,13 +166,29 @@ class ConfigureIntegration extends DeprecatedAsyncView<Props, State> {
     }
   };
 
-  handleOpsgenieMigration = () => {
-    this.setState(
-      {
-        plugins: (this.state.plugins || []).filter(({id}) => id === 'opsgenie'),
-      },
-      () => addSuccessMessage(t("This doesn't do anything yet!"))
-    );
+  handleOpsgenieMigration = async () => {
+    try {
+      const {
+        organization,
+        params: {integrationId},
+      } = this.props;
+
+      await this.api.requestPromise(
+        `/organizations/${organization.slug}/integrations/${integrationId}/migrate-opsgenie/`,
+        {
+          method: 'PUT',
+          data: {},
+        }
+      );
+      this.setState(
+        {
+          plugins: (this.state.plugins || []).filter(({id}) => id === 'opsgenie'),
+        },
+        () => addSuccessMessage(t('Migration in progress.'))
```

Comment:
this will be a little toast at the bottom that goes away after a couple of seconds. do you want it to persist on the screen until the migration is done?

Reply:
I don't think that it needs to, but let me know if you suggest otherwise.

Reply:
what will the user see once the migration has completed?

Reply:
Currently, nothing

Reply:
it would be nice to show some kind of state that tells the user the migration in progress and when it's finished. i realize it's probably not possible to show the live progress because it's an async task. is there any way to let the frontend know the migration has finished?

Reply:
maybe you can disable the plugin once the migration has finished in the UI? but you need some way of knowing if the migration has finished

Reply:
I'll see if I can do that.

Reply:
chatted offline -- it seems difficult to show any kind of progress state. for GA the plugin will be hidden when the migration has completed. we can tell users to refresh to see if the migration has completed, although it's unclear how long the migration will take for an org with a lot of projects